app/questionnaire/navigation.py

```diff
@@ -36,16 +37,18 @@ def build_navigation(self, current_group_id, current_group_instance):
         for group in filter(lambda x: 'hide_in_navigation' not in x, self.survey_json['groups']):
             first_location = Location(group['id'], 0, group['blocks'][0]['id'])
 
-            logger.debug("Building frontend navigation for group %s", group['id'])
+            logger.debug("building frontend navigation", group_id=group['id'])
 
-            repeating_rule = SchemaHelper.get_repeat_rule(group)
             skip_group = self._should_skip_group(current_group_instance, group)
 
             if not skip_group:
+                repeating_rule = SchemaHelper.get_repeat_rule(group)
                 if repeating_rule:
+                    logger.debug("building repeating navigation", group_id=group['id'])
```

Comment:
Are these debug logs delivering much value as the give no context as to what is actually building. e.g. which repeating rule?